enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,454 @@
+> to become tainted. We are tracking and supporting [an effort] in upstream
+> Gateway API to separate experimental into its own group, which we expect to
+> help move towards a better overall experience for users who want experimental
+> Gateway API features going forward.
+
+**We simply cannot anticipate all of the negative effects** succession will
+have on existing implementations if the cluster admin forces through the upgrade
+check. As an extra precaution for users forcing upgrades precipitously, we will
+provide an admin gate to both provide an extra warning and gather consent from
+the cluster admin to take over the management of the CRDs. This will be
+accompanied by detailed documentation on what the admin should check and how to
+fufill the checks, which will also be linked from the description provided in
+the admin gate. An "unsupported config override" will be used to flag
+exceptions.
+
+The admin will be responsible for ensuring the safety of succession. If they
+force through the pre-upgrade check AND the admin gate, the admin gate will be
+left behind to help aid investigation of the cause of problems when the upgrade
+goes wrong.
+
+> **Note**: Even if they force through, we will take over the CRDs once the
+> upgrade completes (or go `Degraded` if that fails for some reason) unless
+> they are in "unsupported config override" mode.
+
+[Group/Version/Kind (GVK)]:https://book.kubebuilder.io/cronjob-tutorial/gvks.html
+[an effort]:https://github.com/kubernetes-sigs/gateway-api/discussions/3497
+
+### Workflow Description
+
+The workflow in this case is an upgrade process. From the _user_ perspective the
+CRDs will be fully managed via the platform from here on out, so they only need
+to interface with the upgrade workflow on the condition that their cluster had
+previously installed Gateway API CRDs. The workflow consists of the pre-upgrade
+checks and the post-upgrade checks.
+
+### Pre-upgrade
+
+1. In the CIO a pre-upgrade check verifies CRD presence
+  * IF the CRDs are present
+    * an admingate is created requiring acknowledgement of CRD succession
+    * UNTIL the schema exactly matches the version we provide we set `Upgradable=false`
+2. Once CRDs are not present OR are an exact match we set `Upgradable=true`
+
+> **Note**: A **cluster-admin** is required for these steps.
+
+> **Note**: The logic for these lives in in the previous release (4.18), but
+> does not need to be carried forward to future releases as other logic exists
+> there to handle Gateway API CRD state (see below).
+
+### Post-upgrade
+
+> **Note**: The logic for these lives in in the new release (4.19) and onward.
+
+1. The CIO is hereafter deployed alongside its CRD protection [Validating Admission Policy (VAP)]
+2. The CIO constantly checks for the presence of the CRDs
+  * IF the CRDs are present
+    * UNTIL there are no unexpected CRDs, the CIO removes any unexpected CRDs
```

Comment:
A VAP is preventing someone from installing these, so to even get to this step, someone has removed the VAP? Which should be auto-recreated right?